shell/common/rasterizer.cc

```diff
@@ -270,13 +271,29 @@ sk_sp<SkImage> Rasterizer::DoMakeRasterSnapshot(
                 return;
               }
 
+              GrDirectContext* context = surface_->GetContext();
+              auto max_size = context->maxRenderTargetSize();
+              double max_scale_factor =
```

Comment:
While the surface is scaled correctly, that scale hasn't been applied to the picture that is going to be drawn onto this surface (should happen in the `DrawSnapshot` call which probably needs a scale parameter or the canvas surface a transformation already pushed onto its CTM to fix this). So, the contents will looked clipped but will have the same aspect ratio. I think a pixel test in the `embedder_unittests` harness will also cover this.

Reply:
Good point -- I corrected this and added fixture check. Lemme know what you think! Based on how some of the other tests work, I don't _think_ it's especially prone to flakiness across test beds (but I could be wrong).